### DISCUSSION TITLE
Gleam language roadmap and "help needed" pointers

### DISCUSSION THREAD

**Discussion by jayber** [gleam-lang/gleam#3983] [Questions & support] [answered]
Hello

I've been looking into Gleam on and off for a while and have started doing Advent of Code in Gleam. 

I wonder if there is a roadmap for future developments in the language or ecosystem? I'd like to know if string templating is coming in the future or patterns in function arguments for example.

On top of that I'd also like to know if there are specific projects or efforts in the Gleam world that people want help with to which I might be able to contribute.

Many thanks for your help and sorry if this question has been answered elsewhere. Just post a link if it has.

**Comment by GearsDatapacks** [comment, marked as answer]
There is a recently released Gleam roadmap! https://gleam.run/roadmap
You can also look at github issues to get an idea of what is planned in future. 
String templating has not been discussed in a long time for Gleam (see: #1473)
Patterns in function arguments are also not planned, and will almost certainly never be implemented as they don't add any new benefit to the user. 
As for other projects needing help, you can look at the [Gleam SaaS](https://github.com/orgs/gleam-lang/projects/5/views/1) software list, which has many libraries which Gleam is currently lacking.
